File: python/submit_job.py
import os
import sys

import requests


API_KEY = os.getenv("MIDJOURNEY_API_KEY")
BASE_URL = os.getenv("MIDJOURNEY_API_BASE_URL", "https://api.midjourney-api.com")
WEBHOOK_URL = os.getenv("MIDJOURNEY_WEBHOOK_URL", "https://yourapp.com/webhooks/midjourney")


def main() -> int:
    if not API_KEY:
        print("Missing MIDJOURNEY_API_KEY")
        return 1

    prompt = " ".join(sys.argv[1:]) or "a futuristic city skyline at sunset --ar 16:9"

    response = requests.post(
        f"{BASE_URL}/midjourney/v1/submit-jobs",
        headers={
            "API-KEY": API_KEY,
            "Content-Type": "application/json",
        },
        json={
            "prompt": prompt,
            "mode": "fast",
            "hookUrl": WEBHOOK_URL,
            "timeout": 600,
        },
        timeout=30,
    )
    response.raise_for_status()
    data = response.json()
    print(data)

    task_id = data.get("data", {}).get("taskId")
    if task_id:
        print(f"\nTask ID: {task_id}")
        print(f"Poll next with: python3 python/poll_status.py {task_id}")

    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: nodejs/submit-job.js
const API_KEY = process.env.MIDJOURNEY_API_KEY;
const BASE_URL =
  process.env.MIDJOURNEY_API_BASE_URL || "https://api.midjourney-api.com";
const WEBHOOK_URL =
  process.env.MIDJOURNEY_WEBHOOK_URL ||
  "https://yourapp.com/webhooks/midjourney";

async function main() {
  if (!API_KEY) {
    throw new Error("Missing MIDJOURNEY_API_KEY");
  }

  const prompt =
    process.argv.slice(2).join(" ") ||
    "a futuristic city skyline at sunset --ar 16:9";

  const response = await fetch(`${BASE_URL}/midjourney/v1/submit-jobs`, {
    method: "POST",
    headers: {
      "API-KEY": API_KEY,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      prompt,
      mode: "fast",
      hookUrl: WEBHOOK_URL,
      timeout: 600,
    }),
  });

  if (!response.ok) {
    throw new Error(`Request failed: ${response.status}`);
  }

  const data = await response.json();
  console.log(JSON.stringify(data, null, 2));

  const taskId = data?.data?.taskId;
  if (taskId) {
    console.log(`\nTask ID: ${taskId}`);
    console.log(`Poll next with: npm run poll -- ${taskId}`);
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
